packages/ens-controller/src/EnsController.ts

```diff
@@ -193,6 +260,74 @@ export class EnsController extends BaseControllerV2<
     });
     return true;
   }
+
+  #getNetworkEnsSupport(network: NetworksChainId) {
+    return Boolean(ensNetworkMap[network]);
+  }
+
+  lookup(ensName: string | Promise<string>): Promise<string | null> {
+    if (!this.ethProvider) {
+      throw new Error('Provider has not been initialized.');
+    }
+    return this.ethProvider.resolveName(ensName);
+  }
+
+  reverse(address: string | Promise<string>): Promise<string | null> {
+    if (!this.ethProvider) {
+      throw new Error('Provider has not been initialized.');
+    }
+    return this.ethProvider.lookupAddress(address);
+  }
+
+  async reverseResolveAddress(nonChecksummedAddress: string) {
+    if (!this.ethProvider) {
+      throw new Error('Provider has not been initialized.');
```

Comment:
The extension implementation of this method will return `undefined` rather than throw in this scenario. It's written such that the caller can attempt reverse resolving an address even if the current chain isn't supported by our ENS controller.

We could consider making this throw instead; maybe that would be a better API (I don't have a preference either way right now). But we'd have to change expectations for the caller; we'd want to ensure this was only called if the current network has been verified as ENS compatible. That would make it more challenging to integrate this controller into the extension. Something to consider.